File: api-server/src/main/java/com/talentlink/talentlink/talentsell/TalentSellRepository.java
package com.talentlink.talentlink.talentsell;

import com.talentlink.talentlink.user.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface TalentSellRepository extends JpaRepository<TalentSell, Long> {
    List<TalentSell> findAllByOrderByCreatedAtDesc();
    // User 객체를 받아 삭제하는 메서드를 추가합니다. (네 파일 모두 동일하게 추가)
    void deleteByUser(User user);
    List<TalentSell> findByUser(User user);
}

File: front-server/app/src/main/java/com/example/talent_link/data/repository/TalentSellRepository.kt
package com.example.talent_link.data.repository

import android.content.Context
import com.example.talent_link.data.model.talentsell.TalentSellResponse
import com.example.talent_link.data.network.RetrofitClient
import com.example.talent_link.util.getToken
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Response

class TalentSellRepository(private val context: Context) {

    private val api = RetrofitClient.talentSellService

    suspend fun uploadTalentSell(
        request: RequestBody,
        image: MultipartBody.Part?
    ): Response<TalentSellResponse> {
        val token = getToken(context) ?: ""
        return api.uploadTalentSell("Bearer $token", request, image)
    }
}
